homeassistant/components/linkplay/__init__.py

```diff
@@ -32,13 +33,26 @@ async def async_setup_entry(hass: HomeAssistant, entry: LinkPlayConfigEntry) ->
     session: ClientSession = await async_get_client_session(hass)
     bridge: LinkPlayBridge | None = None
 
+    # try create a bridge
     try:
         bridge = await linkplay_factory_httpapi_bridge(entry.data[CONF_HOST], session)
     except LinkPlayRequestException as exception:
         raise ConfigEntryNotReady(
             f"Failed to connect to LinkPlay device at {entry.data[CONF_HOST]}"
         ) from exception
 
+    # setup the controller and discover multirooms
+    controller = None
+    if CONTROLLER not in hass.data[DOMAIN]:
```

Comment:
You never actually initialized hass.data[DOMAIN] afaiw, so doesn't this raise a Key error?

Also, could we use HassKey for typing hass.data?

Reply:
I haven't seen the key error during testing, but the default should be set.